notificationdemo/source/android/app/src/main/java/org/kaaproject/kaa/demo/notification/fragment/TopicFragment.java

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright 2014-2016 CyberVision, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kaaproject.kaa.demo.notification.fragment;
+
+import org.kaaproject.kaa.demo.notification.R;
+import org.kaaproject.kaa.demo.notification.MainActivity;
+import org.kaaproject.kaa.demo.notification.entity.TopicPojo;
+import org.kaaproject.kaa.demo.notification.kaa.KaaManager;
+import org.kaaproject.kaa.demo.notification.storage.TopicStorage;
+import org.kaaproject.kaa.demo.notification.adapter.TopicAdapter;
+import org.kaaproject.kaa.demo.notification.util.TopicHelper;
+
+import android.app.Activity;
+import android.content.Context;
+import android.os.Bundle;
+import android.support.v4.app.ListFragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ListView;
+import android.widget.Toast;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+
+/**
+ * Show all topics. You can subscribe on topic and click to see notifications.
+ * Provide timer for updating fragment.
+ * https://developer.android.com/training/basics/fragments/communicating.html
+ */
+public class TopicFragment extends ListFragment implements OnFragmentUpdateEvent,
+        TopicAdapter.OnSubscribeCallback {
+
+    private OnTopicClickedListener mCallback;
+    private Timer timer;
+
+    public TopicFragment() {
+        super();
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public View onCreateView(final LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        getActivity().getActionBar().setTitle(R.string.topic_title);
+
+        updateAdapter();
+
+        timer = new Timer(true);
```

Comment:
I don't understand why you need timer in this application. Looks like this invokes some periodic update of UI. Why this update need to happen periodically? I believe this should happen only if new topic list or notification is received.